src/lib/homestore_backend/tests/hs_shard_tests.cpp

```diff
@@ -162,6 +162,7 @@ TEST_F(ShardManagerTestingRecovery, ShardManagerRecovery) {
     _home_object.reset();
     LOGI("restart home_object");
     _home_object = homeobject::init_homeobject(std::weak_ptr< homeobject::HomeObjectApplication >(app));
+    std::this_thread::sleep_for(std::chrono::seconds{5});
```

Comment:
could we use `pg->repl_dev_->get_leader_id()` to check in L190 and sleep if not the leader?

Reply:
I use 5s because the election_time_out is 3.2s. https://github.com/eBay/nuraft_mesg/blob/7e8cdeb261270642dac186902351da2174d7f077/src/lib/manager_impl.cpp#L25C1-L25C72

i want to refactor this two test case here , since they are much similiar and  have a lot of duplicated logic as
https://github.com/eBay/HomeObject/blob/a0df650e0cfb6391cd0dd6816175bcc8b284e944/src/lib/homestore_backend/tests/hs_blob_tests.cpp#L114

can we do this in a separate refactor PR?